x-pack/plugins/security_solution/public/detections/components/detection_page_filters/index.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback } from 'react';
+import type { Filter } from '@kbn/es-query';
+import { isEqual } from 'lodash';
+import { DEFAULT_DETECTION_PAGE_FILTERS } from '../../../../common/constants';
+import type { FilterGroupProps } from '../../../common/components/filter_group/types';
+import { FilterGroup } from '../../../common/components/filter_group';
+
+type FilterItemSetProps = Omit<FilterGroupProps, 'initialControls'>;
+
+const FilterItemSetComponent = (props: FilterItemSetProps) => {
+  const { dataViewId, onFilterChange, ...restFilterItemGroupProps } = props;
+
+  const [initialFilterControls] = useState(DEFAULT_DETECTION_PAGE_FILTERS);
+
+  const filterChangesHandler = useCallback(
+    (newFilters: Filter[]) => {
+      if (!onFilterChange) {
+        return;
+      }
+      const updatedFilters = newFilters.map((filter) => {
+        return {
+          ...filter,
+          meta: {
+            ...filter.meta,
+            disabled: false,
+          },
+        };
+      });
+      onFilterChange(updatedFilters);
+    },
+    [onFilterChange]
+  );
+  return (
+    <FilterGroup
+      dataViewId={dataViewId}
+      onFilterChange={filterChangesHandler}
+      initialControls={initialFilterControls}
+      {...restFilterItemGroupProps}
+    />
+  );
+};
+
+const arePropsEqual = (prevProps: FilterItemSetProps, newProps: FilterItemSetProps) => {
+  const _isEqual = isEqual(prevProps, newProps);
```

Comment:
@michaelolo24 Yes, props were deep object and because of that it was rendering more than needed. But I do not remeber which prop exactly was creating the issue but I will check and update this comment again.